README.md

```diff
@@ -37,40 +51,236 @@ yarn chain
+---
+
+## Checkpoint 1: Configure Deployment & Wallets
+
+Like the [token vendor challenge](https://speedrunethereum.com/challenge/token-vendor), we'll be building an `Ownable` contract. The contract owner is the **Guru** (the service provider in this application), and you will use multiple browser windows or tabs to assume the roles of Guru and rube (service provider & client).
+
+> 👁 `contract Streamer` inherits `Ownable` with the `is` keyword. `Ownable` comes from [openzeppelin-contracts](https://github.com/OpenZeppelin/openzeppelin-contracts) - a collection of high quality smart contract library code.
+
+> 📝 In `packages/hardhat/deploy/00_deploy_streamer.js`, uncomment the lines of code that deploy the contract and transfer ownership. You will need to enter your own frontend address, to act as Guru.
+
+You'll have to redeploy with `yarn deploy --reset`.
+
+We'll need another active address to act as the rube in our app. To do this just open new tab in your browser.
+
+> ⚠️ **Note**: previous challenges created new addresses by opening an incognito window or a different browser. This will **not** work for this challenge, because the off-chain application uses a very simple communication pipe that doesn't work between different browsers or private windows.)
+
+![wallets-setup](https://github.com/scaffold-eth/se-2-challenges/assets/55535804/1c0ede93-fd7d-4683-a3f0-803b52ca947c)
+
+### 🥅 Goals
+
+- [ ] Does your original frontend address recieve the `Hello Guru` UI?
+- [ ] Does your alternate addresses recieve the `Hello Rube` UI?
+
+---
+
+## Checkpoint 2: Fund a Channel
+
+Like the [decentralized staking challenge](https://speedrunethereum.com/challenge/decentralized-staking), we'll track balances for individual channels / users in a mapping:
+
+```
+mapping (address => uint256) balances;
+```
+
+Rubes seeking wisdom will use a **payable** `fundChannel()` function, which will update this mapping with the supplied balance.
+
+> 📝 Edit `packages/hardhat/contracts/Streamer.sol` to complete the `fundChannel()` function
+
+> 👁 Check `packages/nextjs/pages/streamer.tsx` to see the frontend calling this function. (ctrl-f fundChannel)
+
+![channel-open](https://github.com/scaffold-eth/se-2-challenges/assets/55535804/69a30c45-d384-476f-8b7d-67bc23d21833)
+
+### 🥅 Goals:
+
+- [ ] Does opening a channel (from Rube's tab, you may need some funds from the faucet) cause a `Received Wisdom` box to appear?
+- [ ] Do opened channels appear on the Guru's UI as well?
+- [ ] Using the _Debug Contracts_ tab, does a repeated call to `fundChannel` fail?
+
+---
+
+## Checkpoint 3: Exchange the Service
+
+Now that the channel is funded, and all participants have observed the funding via the emitted event, we can begin our off-chain exchange of service. We are now working in `packages/nextjs/pages/streamer.tsx`.
+
+Functions of note:
+
+- `provideService`: The Guru sends wisdom over the wire to the client.
```

Comment:
Functions ~of~ to note?

Also, I didn't find `provideService` in `streamer.tsx` did we mean to say `provideWisdom` (this is present int `streamer.tsx`) ? 

If we have mistaken, maybe we update `provideWisdom` -> `provideService` in `streamer.tsx` ?